tms/models/tms_team.py

```diff
@@ -0,0 +1,132 @@
+        "tms_order_team_stage_rel",
+        "team_id",
+        "stage_id",
+        string="Stages",
+        default=_default_stages,
+    )
+    order_ids = fields.One2many(
+        "tms.order",
+        "team_id",
+        string="Orders",
+        domain=[("stage_id.is_closed", "=", False)],
+    )
+    order_count = fields.Integer(compute="_compute_order_count", string="Orders Count")
+    sequence = fields.Integer(default=1, help="Used to sort teams. Lower is better.")
+
+    vehicle_ids = fields.One2many("fleet.vehicle", "team_id")
+    vehicle_count = fields.Integer(
+        compute="_compute_vehicle_count", string="Vehicles Count"
+    )
+
+    driver_ids = fields.One2many(
+        "res.partner",
+        "team_id",
+        domain=[("tms_type", "=", "driver")],
+        context={"default_tms_type": "driver"},
+    )
+    driver_count = fields.Integer(
+        compute="_compute_driver_count", string="Drivers Count"
+    )
+
+    crew_ids = fields.One2many("tms.crew", "team_id")
+
+    company_id = fields.Many2one(
+        "res.company",
+        string="Company",
+        required=True,
+        index=True,
+        default=lambda self: self.env.company,
+        help="Company related to this order",
+    )
+
+    trips_todo_count = fields.Integer(
+        string="Number of Trips", compute="_compute_trips_todo_count"
+    )
+    trips_todo_count_draft = fields.Integer(
+        string="Number of Trips", compute="_compute_trips_todo_count"
+    )
+    trips_todo_count_confirmed = fields.Integer(
+        string="Number of Trips", compute="_compute_trips_todo_count"
+    )
+
+    @api.depends("order_ids.stage_id")
+    def _compute_trips_todo_count(self):
+        for team in self:
+            team.order_ids = self.env["tms.order"].search(
+                [
+                    ("team_id", "=", team.id),
+                    (
+                        "stage_id.id",
+                        "!=",
+                        team.env.ref("tms.tms_stage_order_completed").id,
+                    ),
+                ]
+            )
+            data = self.env["tms.order"].read_group(
+                [
+                    ("team_id", "=", team.id),
+                    (
+                        "stage_id.id",
+                        "!=",
+                        team.env.ref("tms.tms_stage_order_completed").id,
+                    ),
+                ],
+                ["stage_id"],
+                ["__count"],
+            )
+            team.trips_todo_count = sum(count for (_, count) in data)
+            team.trips_todo_count_draft = sum(
+                count
+                for (stage_id, count) in data
+                if stage_id.id == team.env.ref("tms.tms_stage_order_draft").id
+            )
+            team.trips_todo_count_confirmed = sum(
+                count
+                for (stage_id, count) in data
+                if stage_id.id == team.env.ref("tms.tms_stage_order_confirmed").id
```

Comment:
This issue of fixed stages in the code doesn't seem to be a good solution. I understand that if that's the case, it's better to define parameter fields in the stage registration.